### PR TITLE
High DPI changes in .NET 4.8 - Remaining documentation

### PR DESCRIPTION
**Pull request by dreddy-work**

### CODE REVIEW

File: Documentation/compatibility/winforms-hdpi-changes-48.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    </configuration>
+```    
+-  High DPI improvements in each major category has its own setting to opt in. Applications can set the required setting by following instructions explained at [High DPI support in Windows Forms](https://docs.microsoft.com/en-us/dotnet/framework/winforms/high-dpi-support-in-windows-forms).
+
+Note that many of these improvements are enabled only on windows 10 and your application must be enabled to run in Windows 10 compatible mode. You can set this in the application manifest file as defined in [High DPI support in Windows Forms](https://docs.microsoft.com/en-us/dotnet/framework/winforms/high-dpi-support-in-windows-forms)
+
+Applications that target the .NET Framework 4.8 and want to preserve the legacy high dpi behavior can opt in to the use of legacy behaviour by explicitly setting the required switch to `true`.
+
+__Improved PropertyGrid scaling when DPI changed.__
+- Improved scaling of PropertyGrid and its child controls when the DPI of of the application is changed ( Application moved from one monitor to another with different DPI settings). All changes made for this improvement are quirked under switch `DisableDpiChangedHighDpiImprovements`.
```
Comment (OliaG) [inline Documentation/compatibility/winforms-hdpi-changes-48.md]:
```suggestion
- Improved scaling of PropertyGrid and its child controls when the DPI of the application is changed (application moved from one monitor to another with different DPI settings). All changes made for this improvement are quirked under switch `DisableDpiChangedHighDpiImprovements`.
```

File: Documentation/compatibility/winforms-hdpi-changes-48.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+__Improved scaling of of Toolstrip Menus/Buttons/Glyphs when the DPI of the application changed__
```
Comment (OliaG) [inline Documentation/compatibility/winforms-hdpi-changes-48.md]:
```suggestion
__Improved scaling of Toolstrip Menus/Buttons/Glyphs when the DPI of the application changed__
```

File: Documentation/compatibility/winforms-hdpi-changes-48.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+- Improved the scaling of Toolstrip Menus/Buttons/Glyphs when the DPI of the application changed. All changes made for this improvement are quirked under switch `ToolStrip.DisableHighDpiImprovements` & `DisableDpiChangedHighDpiImprovements`.
+
+__Improved scaling of RadioButton and CheckBox when application DPI changed__
+- Improved the scaling of the RadioButton and CheckBox , when setting different styles on them and the DPI of the application changed. All changes made for this improvement are quirked under switch `DisableDpiChangedHighDpiImprovements`.
```
Comment (OliaG) [inline Documentation/compatibility/winforms-hdpi-changes-48.md]:
```suggestion
- Improved the scaling of the RadioButton and CheckBox, when setting different styles on them and the DPI of the application changed. All changes made for this improvement are quirked under switch `DisableDpiChangedHighDpiImprovements`.
```

File: Documentation/compatibility/winforms-hdpi-changes-48.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+### Version Introduced
+4.8
+
+### Source Analyzer Status
+NotPlanned
+
+### Change Description
+The Windows Forms Framework is continuing to improve the scaling and rendering of its controls to better support Windows Forms customers on high resolution devices. These include the following changes:
+- Changes to improve the scaling of PropertyGrid when the DPI of the application changed.
```
Comment (OliaG) [inline Documentation/compatibility/winforms-hdpi-changes-48.md]: Can we add an explanation of "when the DPI of the application changed" like we change monitor setting while the application is running or we drag an application to another monitor with a different DPI setting

File: Documentation/compatibility/winforms-hdpi-changes-48.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+- Changes to improve the scaling of Font based controls when the DPI of the application changed.
+- Changes to improve the scaling of Button control when the DPI of the application changed.
+
+- [x] Quirked
+- [ ] Build-time break
+
+### Recommended Action
+__How to opt in or out of these changes__
+  
+In order for the application to benefit from these changes, it must run on the .NET Framework 4.8. The application can opt in into these changes in either of the following ways:
```
Comment (OliaG) [inline Documentation/compatibility/winforms-hdpi-changes-48.md]:
```suggestion
__Opting In

In order for the application to benefit from these changes, it must either target .NET Framework 4.8. or be explicitly enabled in application targeting earlier versions of .NET Framework.

For Framework 4.8 make sure your application has app.manifest file (if not - add it) and in that file Windows 10 compatible mode is enabled as it is specified here: [High DPI support in Windows Forms](https://docs.microsoft.com/en-us/dotnet/framework/winforms/high-dpi-support-in-windows-forms).
```

File: Documentation/compatibility/winforms-hdpi-changes-48.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+- It is recompiled to target the .NET Framework 4.8. These high DPI improvement changes are enabled by default on Windows Forms applications that target the .NET Framework 4.8.
```
Comment (OliaG) [inline Documentation/compatibility/winforms-hdpi-changes-48.md]:
```suggestion

```

File: Documentation/compatibility/winforms-hdpi-changes-48.md [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+- It targets the .NET Framework 4.7.2 or earlier version and opts in to this new behaviour by adding the following [Windows Forms Configuration Section](https://docs.microsoft.com/en-us/dotnet/framework/configure-apps/file-schema/winforms/) to the app config file and then the required flag to opt-in specific feature improvements.
```
Comment (OliaG) [inline Documentation/compatibility/winforms-hdpi-changes-48.md]:
```suggestion
For .NET Framework 4.7.2 or earlier version to opts in to this new behavior add the following [Windows Forms Configuration Section](https://docs.microsoft.com/en-us/dotnet/framework/configure-apps/file-schema/winforms/) to the app.config file and then the required flag to opt-in specific feature improvements.
```

File: Documentation/compatibility/winforms-hdpi-changes-48.md [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+      <configuration>
+        <System.Windows.Forms.ApplicationConfigurationSection>
+        ...
+        </System.Windows.Forms.ApplicationConfigurationSection>
+      </configuration>
+    </configuration>
+```    
+-  High DPI improvements in each major category has its own setting to opt in. Applications can set the required setting by following instructions explained at [High DPI support in Windows Forms](https://docs.microsoft.com/en-us/dotnet/framework/winforms/high-dpi-support-in-windows-forms).
+
+Note that many of these improvements are enabled only on windows 10 and your application must be enabled to run in Windows 10 compatible mode. You can set this in the application manifest file as defined in [High DPI support in Windows Forms](https://docs.microsoft.com/en-us/dotnet/framework/winforms/high-dpi-support-in-windows-forms)
```
Comment (OliaG) [inline Documentation/compatibility/winforms-hdpi-changes-48.md]:
```suggestion
```

File: Documentation/compatibility/winforms-hdpi-changes-48.md [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+
```
Comment (OliaG) [inline Documentation/compatibility/winforms-hdpi-changes-48.md]:
```suggestion

__Opting out
```

File: Documentation/compatibility/winforms-hdpi-changes-48.md [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+Applications that target the .NET Framework 4.8 and want to preserve the legacy high dpi behavior can opt in to the use of legacy behaviour by explicitly setting the required switch to `true`.
```
Comment (OliaG) [inline Documentation/compatibility/winforms-hdpi-changes-48.md]:
```suggestion
Applications that target the .NET Framework 4.8 and want to preserve the legacy high dpi behavior can do so by explicitly setting to `true` the switches that will be described bellow.
```

File: Documentation/compatibility/winforms-hdpi-changes-48.md (hunk 3 above)
Comment (rpetrusha) [inline Documentation/compatibility/winforms-hdpi-changes-48.md]:
```suggestion
All changes made in this area are controlled by the `DisableDpiChangedHighDpiImprovements` switch.
```

File: Documentation/compatibility/winforms-hdpi-changes-48.md [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+
+__Improved scaling of of Toolstrip Menus/Buttons/Glyphs when the DPI of the application changed__
+- Improved the scaling of Toolstrip Menus/Buttons/Glyphs when the DPI of the application changed. All changes made for this improvement are quirked under switch `ToolStrip.DisableHighDpiImprovements` & `DisableDpiChangedHighDpiImprovements`.
+
+__Improved scaling of RadioButton and CheckBox when application DPI changed__
+- Improved the scaling of the RadioButton and CheckBox , when setting different styles on them and the DPI of the application changed. All changes made for this improvement are quirked under switch `DisableDpiChangedHighDpiImprovements`.
+
+__Improved scaling of Button control when application DPI changed__
+- Improved the scaling of the Button control when the DPI of the application changed. All changes made for this improvement are quirked under switch `DisableDpiChangedHighDpiImprovements`.
```
Comment (rpetrusha) [inline Documentation/compatibility/winforms-hdpi-changes-48.md]:
```suggestion
All changes made in this area are controlled by the  `DisableDpiChangedHighDpiImprovements` switch.
```

File: Documentation/compatibility/winforms-hdpi-changes-48.md [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+
+__Improved scaling of Font based controls when application DPI changed__
```
Comment (rpetrusha) [inline Documentation/compatibility/winforms-hdpi-changes-48.md]:
```suggestion
__Improved scaling of Font based controls when application DPI changes__
```

File: Documentation/compatibility/winforms-hdpi-changes-48.md [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+ Improved the scaling of Font based controls when the DPI of the application changed by scaling font on them. All changes made for this improvement are quirked under switch `DisableDpiChangedHighDpiImprovements`.
```
Comment (rpetrusha) [inline Documentation/compatibility/winforms-hdpi-changes-48.md]:
```suggestion
All changes made in this area are controlled by the `DisableDpiChangedHighDpiImprovements` switch.
```

File: Documentation/compatibility/winforms-hdpi-changes-48.md [hunk 14]
```diff
@@ -0,0 +1,84 @@
...
+ 
+### Affected APIs 
+
```
Comment (rpetrusha) [inline Documentation/compatibility/winforms-hdpi-changes-48.md]:
```suggestion
- Not detectable via API analysis.
```

File: Documentation/compatibility/winforms-hdpi-changes-48.md [hunk 15]
```diff
@@ -0,0 +1,84 @@
...
+Major
+
+### Version Introduced
+4.8
+
+### Source Analyzer Status
+NotPlanned
+
+### Change Description
+The Windows Forms Framework is continuing to improve the scaling and rendering of its controls to better support Windows Forms customers on high resolution devices. These include the following changes:
```
Comment (rpetrusha) [inline Documentation/compatibility/winforms-hdpi-changes-48.md]:
```suggestion
The Windows Forms Framework is continuing to improve the scaling and rendering of its controls to better support Windows Forms customers on high resolution devices. As a result of these improvemeents, Windows Forms application are able to better handle DPI changes that occur, for example, when moving the application from one monitor to another monitor that has different DPI settings. These include the following changes:
```

**Comment by sujayvsarma** [comment]
Would have been a good practice to push all the docs in one go instead of so many. Folks that had subscribed to this repo with PR notifications would have been heavily spammed.

**Comment by rpetrusha** [comment]
Thanks for making the additional changes, @dreddy-work. I've also made a few minor changes, and I'll merge your PR now.

**Pull request merged**